pipeline/pipeline/backend.py

```diff
@@ -90,5 +111,151 @@ def _get_random_name():
 
         return _get_random_name()
 
-    def copy(self, src, dest):  # FIXME: symbolic links? support gsutil?
-        return f"cp {src} {dest}"
+
+class BatchBackend(Backend):
+    def __init__(self, url):
+        self._batch_client = batch.client.BatchClient(url)
+
+    def run(self, pipeline, dry_run, verbose, delete_scratch_on_exit):  # pylint: disable-msg=R0915
+        if dry_run:
+            raise NotImplementedError
+
+        hail_scratch_bucket = 'hail-pipeline-scratch'
+        subdir_name = 'pipeline-{}'.format(uuid.uuid4().hex)
+
+        remote_tmpdir = f'gs://{hail_scratch_bucket}/{subdir_name}/'
+        local_tmpdir = '/tmp/pipeline/'
+
+        default_image = 'ubuntu'
+
+        batch = self._batch_client.create_batch()
+        n_jobs_submitted = 0
+
+        task_to_job_mapping = {}
+        job_id_to_command = {}
+
+        volumes = [{'volume': {'name': 'pipeline-test-0-1--hail-is-service-account-key',
+                               'secret': {'optional': False,
+                                          'secretName': 'pipeline-test-0-1--hail-is-service-account-key'}},
+                    'volume_mount': {'mountPath': '/secrets',
+                                     'name': 'pipeline-test-0-1--hail-is-service-account-key',
+                                     'readOnly': True}}]
+
+        activate_service_account = 'gcloud auth activate-service-account ' \
+                                   'pipeline-test-0-1--hail-is@hail-vdc.iam.gserviceaccount.com ' \
+                                   '--key-file /secrets/pipeline-test-0-1--hail-is.key'
+
+        for task in pipeline._tasks:
+            def copy_input(r):
+                if isinstance(r, ResourceFile):
+                    if isinstance(r, InputResourceFile):
+                        return [f'gsutil cp {r._input_path} {local_tmpdir}/{r.file_name}']  # FIXME: This is serial!
```

Comment:
I don't think this is a FIXME? If it is let's make an issue instead and assign it to you.